atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -480,6 +481,34 @@ public void testTransactionIsolation() throws Exception {
         }
     }
 
+    @Test
+    public void testGetRowsLocalWritesWithColumnSelection() {
+        // This test ensures getRows correctly applies columnSelection
+        byte[] row1 = PtBytes.toBytes("row1");
+        Cell row1Column1 = Cell.create(row1, PtBytes.toBytes("column1"));
+        Cell row1Column2 = Cell.create(row1, PtBytes.toBytes("column2"));
+        byte[] row1Column1Value = BigInteger.valueOf(1).toByteArray();
+        byte[] row1Column2Value = BigInteger.valueOf(2).toByteArray();
+
+        Transaction snapshotTx = serializableTxManager.createNewTransaction();
```

Comment:
The CachingTransaction seems to hide the bug. The value will be present in ``CachingTransaction#getCachedCellIfPresent``, which will pass an empty ``Set<byte[]> toLoad`` to the delegate ``SnapshotTransaction``.

For some reason the unwrap method at the bottom didn't work for me either :( so I'm pulling in the serializableTxManager.